### PR TITLE
Fix persisting of Catalog Promotion data on create

### PR DESCRIPTION
**Pull request by arti0090**
| Q               | A
| --------------- | -----
| Branch?         |  master 
| Bug fix?        | kind of (fix bug for feature)
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

File: features/promotion/applying_catalog_promotions/applying_catalog_promotions_for_variant.feature [hunk 1]
```diff
@@ -13,9 +13,8 @@ Feature: Applying catalog promotions for variants
         And the store has a "Mug" configurable product
         And this product has "PHP Mug" variant priced at "$5.00"
         And this product has "Kotlin Mug" variant priced at "$7.00"
-        And there is a catalog promotion with "winter_sale" code and "Winter sale" name
-        And it will be applied on "PHP T-Shirt" variant
-        And it will reduce price by "30%"
+        And there is a catalog promotion with "Winter sale" name that will be applied on "PHP T-shirt" variant and will reduce price by "30%"
+
```
Comment (Tomanhez) [inline features/promotion/applying_catalog_promotions/applying_catalog_promotions_for_variant.feature:17]: Not needed blank line

File: src/Sylius/Behat/Context/Setup/CatalogPromotionContext.php [hunk 2]
```diff
@@ -114,6 +115,33 @@ public function itWillReducePrice(CatalogPromotionInterface $catalogPromotion, f
...
+        string $name,
+        ProductVariantInterface $variant,
+        float $discount
+    ): void {
+        $catalogPromotion = $this->createCatalogPromotion($name);
+
+        /** @var CatalogPromotionRuleInterface $catalogPromotionRule */
+        $catalogPromotionRule = $this->catalogPromotionRuleFactory->createNew();
+        $catalogPromotionRule->setType(CatalogPromotionRuleInterface::TYPE_FOR_VARIANTS);
+        $catalogPromotionRule->setConfiguration([$variant->getCode()]);
```
Comment (SirDomin) [inline src/Sylius/Behat/Context/Setup/CatalogPromotionContext.php:131]: i think we can wait with this one, untill https://github.com/Sylius/Sylius/pull/13066 is merged, so we can use 'variants' => [$variant->getCode()]

File: features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotions_on_variants_list.feature [hunk 3]
```diff
@@ -10,9 +10,7 @@ Feature: Seeing applied catalog promotions on variants list
         And this product has "PHP T-Shirt" variant priced at "$20.00"
         And this product has "Kotlin T-Shirt" variant priced at "$40.00"
         And this product has "Python T-Shirt" variant priced at "$40.00"
-        And there is a catalog promotion with "winter_sale" code and "Winter sale" name
-        And it will be applied on "PHP T-Shirt" variant
-        And it will reduce price by "50%"
```
Comment (lchrusciel) [inline features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotions_on_variants_list.feature:15]: The previous steps should be removed I suppose

File: features/promotion/applying_catalog_promotions/applying_catalog_promotions_for_variant.feature [hunk 4]
```diff
@@ -13,9 +13,8 @@ Feature: Applying catalog promotions for variants
         And the store has a "Mug" configurable product
         And this product has "PHP Mug" variant priced at "$5.00"
         And this product has "Kotlin Mug" variant priced at "$7.00"
-        And there is a catalog promotion with "winter_sale" code and "Winter sale" name
-        And it will be applied on "PHP T-Shirt" variant
-        And it will reduce price by "30%"
+        And there is a catalog promotion with "Winter sale" name that will be applied on "PHP T-shirt" variant and will reduce price by "30%"
```
Comment (GSadee) [inline features/promotion/applying_catalog_promotions/applying_catalog_promotions_for_variant.feature:16]:
```suggestion
        And there is a catalog promotion "Winter sale" that applies on "PHP T-shirt" variant and reduces price by "30%"
```

File: features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotions_on_variants_list.feature [hunk 5]
```diff
@@ -10,9 +10,7 @@ Feature: Seeing applied catalog promotions on variants list
...
+        And there is a catalog promotion with "Winter sale" name that will be applied on "PHP T-shirt" variant and will reduce price by "50%"
```
Comment (GSadee) [inline features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotions_on_variants_list.feature:13]:
```suggestion
        And there is a catalog promotion "Winter sale" that applies on "PHP T-shirt" variant and reduces price by "50%"
```

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
But there are still implementation of these steps:
```
        And there is a catalog promotion with "winter_sale" code and "Winter sale" name
        And it will be applied on "PHP T-Shirt" variant
        And it will reduce price by "50%"
```
In behat contexts. Do we want to keep them?

**Comment by lchrusciel** [comment]
Thank you, @arti0090! :tada:

**Pull request merged**